File: Polimorfismo/Ejercicio10/Python/Pokedex.py
class Agua:
    def __init__(self, nombre, ptsvida, nivel):
        self.nombre = nombre
        self.ptsvida = ptsvida
        self.nivel = nivel
    def atacar(self):
        print(self.nombre + " lanza un ataque de Agua")
    def ataque1(self, daño):
        self.ptsvida -= daño
    def ataque2(self):
        self.ptsvida -= 5
class Fuego:
    def __init__(self, nombre, ptsvida, nivel, quemadura):
        self.nombre = nombre
        self.ptsvida = ptsvida
        self.nivel = nivel
        self.quemadura = quemadura
    def atacar(self):
        print(self.nombre + " lanza una llamarada(no la de mou)")
    def atacarConNombre(self, ataque):
        print(self.nombre + " usa " + ataque)
    def ataque1(self, daño):
        self.ptsvida -= daño
    def ataque2(self):
        self.ptsvida -= 5
class Planta:
    def __init__(self, nombre, ptsvida):
        self.nombre = nombre
        self.ptsvida = ptsvida
    def atacar(self):
        print(self.nombre + " usa látigo cepa")
    def ataque1(self, daño):
        self.ptsvida -= daño
    def ataque2(self):
        self.ptsvida -= 5
squirtle = Agua("Squirtle", 100, 5)
totodile = Agua("Totodile", 95, 6)
charmander = Fuego("Charmander", 90, 5, 10)
vulpix = Fuego("Vulpix", 85, 4, 8)
bulbasaur = Planta("Bulbasaur", 100)
greninja = Planta("Greninja", 90)
squirtle.atacar()
charmander.atacarConNombre("Lanzallamas")
bulbasaur.ataque1(20)
vulpix.ataque2()
print("Vida de Bulbasaur:", bulbasaur.ptsvida)
print("Vida de Vulpix:", vulpix.ptsvida)
print("Vida de Gremimja:", greninja.ptsvida)

File: Polimorfismo/Ejercicio10/Java/Pokedex.java
class Agua {
    private String nombre;
    private int puntosdevida;
    private int nivel;
    public Agua(String nombre, int puntosdevida, int nivel) {
        this.nombre = nombre;
        this.puntosdevida = puntosdevida;
        this.nivel = nivel;
    }
    public void recibirAtaque(int daño) {
        puntosdevida -= daño;
    }
    public void recibirAtaque() {
        puntosdevida -= 5;
    }
    public void atacar() {
        System.out.println(nombre + " lanza un ataque de burbuja");
    }
}
class Fuego {
    private String nombre;
    private int puntosdevida;
    private int nivel;
    private int quemadura;
    public Fuego(String nombre, int puntosdevida, int nivel, int quemadura) {
        this.nombre = nombre;
        this.puntosdevida = puntosdevida;
        this.nivel = nivel;
        this.quemadura = quemadura;
    }
    public void recibirAtaque(int daño) {
        puntosdevida -= daño;
    }
    public void recibirAtaque() {
        puntosdevida -= 5;
    }
    public void atacar() {
        System.out.println(nombre + " lanza una llamarada");
    }
    public void atacar(String ataque) {
        System.out.println(nombre + " usa " + ataque);
    }
}
class Planta {
    private String nombre;
    private int puntosdevida;
    private int nivel;
    public Planta(String nombre, int puntosdevida, int nivel) {
        this.nombre = nombre;
        this.puntosdevida = puntosdevida;
        this.nivel = nivel;
    }
    public void recibirAtaque(int daño) {
        puntosdevida -= daño;
    }

    public void recibirAtaque() {
        puntosdevida -= 5;
    }
    public void atacar() {
        System.out.println(nombre + " usa látigo cepa");
    }
}